fuzz/fuzz_targets/decompress_chunked.rs

```diff
@@ -1,133 +1,128 @@
+    // iterate over chunks in the compressed data
+    'decompression: for chunk in fuzz_data.chunks(chunk_size) {
+        // set new chunk as input
+        stream.next_in = chunk.as_ptr() as *mut core::ffi::c_char;
         stream.avail_in = chunk.len() as _;
 
-        let err = unsafe { BZ2_bzDecompress(&mut stream) };
-        match err {
-            BZ_OK => continue,
-            BZ_RUN_OK => panic!("BZ_RUN_OK"),
-            BZ_FLUSH_OK => panic!("BZ_FLUSH_OK"),
-            BZ_FINISH_OK => panic!("BZ_FINISH_OK"),
-            BZ_STREAM_END => continue,
-            BZ_SEQUENCE_ERROR => panic!("BZ_SEQUENCE_ERROR"),
-            BZ_PARAM_ERROR => panic!("BZ_PARAM_ERROR"),
-            BZ_MEM_ERROR => panic!("BZ_MEM_ERROR"),
-            BZ_DATA_ERROR => panic!("BZ_DATA_ERROR"),
-            BZ_DATA_ERROR_MAGIC => panic!("BZ_DATA_ERROR_MAGIC"),
-            BZ_IO_ERROR => panic!("BZ_IO_ERROR"),
-            BZ_UNEXPECTED_EOF => panic!("BZ_UNEXPECTED_EOF"),
-            BZ_OUTBUFF_FULL => panic!("BZ_OUTBUFF_FULL"),
-            BZ_CONFIG_ERROR => panic!("BZ_CONFIG_ERROR"),
-            _ => panic!("{err}"),
+        // process this chunk
+        loop {
+            // perform decompression
+            match unsafe { BZ2_bzDecompress(&mut stream) } {
+                BZ_STREAM_END => {
+                    // stream complete, stop processing chunks
+                    break 'decompression;
+                }
+                BZ_OK => {
+                    // there is still input from this chunk left to process but no more output buffer
+                    // this means we have to increase the output buffer and retry the decompress
+                    if stream.avail_in > 0 && stream.avail_out == 0 {
+                        let used = output.len() - stream.avail_out as usize;
+                        // The dest buffer is full, resize it
+                        let add_space: u32 = Ord::max(4096, output.len().try_into().unwrap());
+                        output.resize(output.len() + add_space as usize, 0);
+
+                        // If resize() reallocates, it may have moved in memory
+                        stream.next_out = output.as_mut_ptr().cast::<i8>().wrapping_add(used);
+                        stream.avail_out += add_space;
+
+                        continue;
+                    } else if stream.avail_in == 0 {
+                        // stop processing this chunk
+                        break;
+                    }
+
+                    continue;
+                }
+                BZ_DATA_ERROR | BZ_DATA_ERROR_MAGIC => {
+                    // valid reasons for decompression errors if the input is invalid
+
+                    // clean up state
+                    unsafe { BZ2_bzDecompressEnd(&mut stream) };
+
+                    // stop processing
+                    return invalid_input;
+                }
+                BZ_FLUSH_OK => {
+                    panic!("BZ_FLUSH_OK");
```

Comment:
```suggestion
                    panic!("BZ_FLUSH_OK")
```

that will makes hits format more compactly

Reply:
I think I've addressed this in d5b6d7295a3ac321c059c1c39e09036ceea61697, but please double-check if I understood you correctly.